### PR TITLE
tooling: Add project release/dev/sync actions

### PR DESCRIPTION
**Pull request by phlax**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: tools/project/README.md [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+--- /dev/null
++++ b/changelogs/1.21.2.yaml
+@@ -0,0 +1,20 @@
++date: April 27, 2022
++
++behavior_changes: []
++
++minor_behavior_changes:
++- area: cryptomb
++  change: |-
```
Comment (phlax) [inline tools/project/README.md:69]: argg this is wrong - i need to fix

File: tools/project/README.md [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+- create PR with committed changes
+- wait for PR to land on `release/v1.23`
+- continue with development
+
+### Sync a release branch
+
+This can be done either on `main` or on another release branch, and would be typically done after an
+_earlier_ release branch had made a release.
+
+For example, supposing `release/v1.21` had just released `v1.23.7`, you might want to include
```
Comment (phlax) [inline tools/project/README.md:305]: s/1.23/1.21/

File: docs/BUILD [hunk 3]
```diff
@@ -179,11 +178,11 @@ genrule(
...
     outs = ["version_histories.tar"],
     cmd = """
-    $(location //tools/docs:generate_version_histories) \
-        %s \
-        $@
-    """ % VERSION,
+    $(location //tools/docs:generate_version_histories) $@
+    """,
     tools = [
+        ":versions.yaml",
```
Comment (phlax) [inline docs/BUILD:184]: not sure if this is necessary

File: tools/docs/generate_version_histories.py [hunk 4]
```diff
@@ -3,7 +3,6 @@
 import sys
 import tarfile
 from functools import cached_property
-
```
Comment (phlax) [inline tools/docs/generate_version_histories.py:6]: leave ws

File: tools/project/README.md [hunk 5]
```diff
@@ -0,0 +1,318 @@
...
+index a6c2798a48..573ce34a59 100644
+--- a/VERSION.txt
++++ b/VERSION.txt
+@@ -1 +1 @@
+-1.23.0
++1.24.0-dev
+
+```
+
+#### Example: open (`release/vX.Y`) branch to development
```
Comment (phlax) [inline tools/project/README.md:206]: as its an example - probs use `release/v1.23`

File: tools/project/README.md [hunk 6]
```diff
@@ -0,0 +1,318 @@
...
+
+For non-`main` release branches this command can be called with the `--patch` option. In this case only the patch version will be
+incremented, rather than the minor version.
+
+```console
```
Comment (phlax) [inline tools/project/README.md:211]: perhaps add `git` commands to checkout the branch, and fork it

File: tools/project/README.md [hunk 7]
```diff
@@ -0,0 +1,318 @@
...
+ changelogs/current.yaml      | 109 +++++--------------------------------------------------------------------------------------------------------
+ 3 files changed, 122 insertions(+), 105 deletions(-)
+
+```
+
+But, in this case, the *patch* version is incremented instead.
+
+```console
+diff --git a/VERSION.txt b/VERSION.txt
+diff --git a/VERSION.txt b/VERSION.txt
```
Comment (phlax) [inline tools/project/README.md:248]: repeat

File: docs/BUILD (hunk 3 above)
Comment (phlax) [reply docs/BUILD:184]: it is

**Review by mattklein123 — CHANGES_REQUESTED** [review]
Amazing!!! cc @envoyproxy/envoy-maintainers 

Can you update the instructions here? https://github.com/envoyproxy/envoy/blob/main/RELEASES.md#cutting-a-major-release

/wait

**Review by mattklein123 — APPROVED** [review]
Amazing!!!

**Pull request merged**